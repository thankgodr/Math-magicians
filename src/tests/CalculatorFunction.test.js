import renderer from 'react-test-renderer';
import { render, queryByAttribute } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import CalculatorFunction from '../components/CalucalatorFunction';

describe('Testing interaction and snaption of calculator button', () => {
  test('Calculator function button snapshot', () => {
    const elem = renderer.create(<CalculatorFunction />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  test('Button is clickable', () => {
    let count = 0;
    const res = () => { count += 1; };
    const dom = render(<CalculatorFunction name="+" add={res} clclassName = "tess" id="plus" />);
    const getById = queryByAttribute.bind(null, 'id');
    const first = getById(dom.container, 'plus');
    userEvent.click(first);
    userEvent.click(first);
    expect(count).toBe(2);
  });
});