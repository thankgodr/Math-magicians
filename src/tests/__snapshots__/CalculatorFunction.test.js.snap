// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing interaction and snaption of calculator button Calculator function button snapshot 1`] = `
<div
  className="col-md-3 function"
  onClick={[Function]}
>
  <p
    className="text-center"
  >
     
    =
  </p>
</div>
`;
